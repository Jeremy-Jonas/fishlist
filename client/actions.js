import request from 'superagent'

export function getFishPending () {
  return {
    type: 'GET_FISH_PENDING'
  }
}

export function getFishSuccess (fish) {
  return {
    type: 'GET_FISH_SUCCESS',
    fish
  }
}

export function getFishDetailSuccess (fishDetail) {
  return {
    type: 'GET_FISH_DETAIL',
    fishDetail
  }
}

export function getFishError (message) {
  return {
    type: 'GET_FISH_ERROR',
    message
  }
}

export function getFish () {
  return dispatch => {
    dispatch(getFishPending())

    request
      .get('/api/v1/fish')
      .then(res => dispatch(getFishSuccess(res.body)))
      .catch(err => dispatch(getFishError(err.message)))
  }
}

export function getFishDetail (fishId) {
  return dispatch => {
    dispatch(getFishPending())

    request
      .get(`/api/v1/fish/${fishId}`)
      .then(res => dispatch(getFishDetailSuccess(res.body)))
      .catch(err => dispatch(getFishError(err.message)))
  }
}
