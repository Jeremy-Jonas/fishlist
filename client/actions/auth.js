import request from 'superagent'

import { setToken } from '../utils/tokens'

export const loginPending = () => ({ type: 'LOGIN_PENDING' })

export const loginSuccess = () => ({ type: 'LOGIN_SUCCESS' })

export const loginFailure = error => ({ type: 'LOGIN_FAILURE', error })

export const login = (username, password) => dispatch => {
  dispatch(loginPending())

  request
    .post('/api/v1/auth/login')
    .send({ username, password })
    .then(res => {
      setToken(res.body.token)
      dispatch(loginSuccess())
    })
    .catch(err => dispatch(loginFailure(err.response.body.error)))
}

export const registerPending = () => ({ type: 'REGISTER_PENDING' })

export const registerSuccess = () => ({ type: 'REGISTER_SUCCESS' })

export const registerFailure = error => ({ type: 'REGISTER_FAILURE', error })

export const register = (username, password, email) => dispatch => {
  dispatch(registerPending())

  return request
    .post('/api/v1/auth/register')
    .send({ username, password, email })
    .then(res => {
      setToken(res.body.token)
      dispatch(registerSuccess())
    })
    .catch(err => dispatch(registerFailure(err.response.body.error)))
}

export const logout = () => ({ type: 'LOGOUT' })
  
