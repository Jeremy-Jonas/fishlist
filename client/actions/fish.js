import request from 'superagent'

export function getFishPending () {
  return {
    type: 'GET_FISH_PENDING'
  }
}

export function getFishSuccess (fish) {
  return {
    type: 'GET_FISH_SUCCESS',
    fish
  }
}

export function getFishDetailSuccess (fishDetail) {
  return {
    type: 'GET_FISH_DETAIL_SUCCESS',
    fishDetail
  }
}

export function getFishError (message) {
  return {
    type: 'GET_FISH_ERROR',
    message
  }
}

export function orderAZ (sortOrder) {
  return {
    type: 'SORT_AZ',
    sortOrder
  }
}

export function orderZA (sortOrder) {
  return {
    type: 'SORT_ZA',
    sortOrder
  }
}

export function getFish () {
  return dispatch => {
    dispatch(getFishPending())

    request
      .get('/api/v1/fish')
      .then(res => dispatch(getFishSuccess(res.body)))
      .catch(err => dispatch(getFishError(err.message)))
  }
}

export function getFishDetail (fishId) {
  return dispatch => {
    dispatch(getFishPending())

    request
      .get(`/api/v1/fish/${fishId}`)
      .then(res => dispatch(getFishDetailSuccess(res.body)))
      .catch(err => dispatch(getFishError(err.message)))
  }
}

export function searchFish (searchedFish) {
  return {
    type: 'SEARCH_FISH',
    searchedFish
  }
}
