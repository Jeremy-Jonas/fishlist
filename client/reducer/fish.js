export default function fishReducer (state = [], action) {
  switch (action.type) {
    case 'GET_FISH_SUCCESS':
      return action.fish

    default:
      return state
  }
}
