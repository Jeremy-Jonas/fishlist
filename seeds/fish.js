exports.seed = knex =>
  knex('fish').del()
    .then(() =>
      knex('fish').insert([
        { name: 'paua' },
        { name: 'cockle' },
        { name: 'kingfish' },
        { name: 'kahawai' },
        { name: 'grey mullet' },
        { name: 'butterfish' },
        { name: 'striped marlin' },
        { name: 'tarakihi' },
        { name: 'whitebait' },
        { name: 'blue cod' },
        { name: 'elephantfish' },
        { name: 'red snapper' }
      ]))
