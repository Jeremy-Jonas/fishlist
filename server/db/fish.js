const connection = require('./')

function get (db = connection) {
  return db('fish')
    .join('levels', 'fish.level_id', '=', 'levels.id')
    .select()
}

function createFish (
  newFish,
  fishDetail,
  db = connection
) {
  return db('fish')
    .insert(newFish)
    .then((res) => db('fish_details').insert({
      ...fishDetail,
      fish_id: res[0]
    }))
}

function updateFish (db, req, res) {
  const fishId = Number(req.params.id)
  return db('fish')
    .where('fishId', fishId)
    .update(fishDetails => res.status(200).json(fishDetails))
}

module.exports = {
  get,
  createFish,
  updateFish
}
