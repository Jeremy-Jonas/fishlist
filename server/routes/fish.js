const express = require('express')

const db = require('../db')

const router = express.Router()

router.get('/', (req, res) => {
  db.getFish()
    .then(fish => res.json(fish))
})

module.exports = router
