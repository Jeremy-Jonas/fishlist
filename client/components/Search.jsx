import React from 'react'
import {Input} from 'semantic-ui-react'
import {connect} from 'react-redux'

import {searchFish} from '../actions/fish'

export class Search extends React.Component {
  render () {
    return (
      <div>
        <Input icon='search' placeholder='Start typing to filter...' onKeyUp={this.props.searchFish} />
      </div>
    )
  }
}

const matchDispatchToProps = (dispatch) => ({
  searchFish: (e) => dispatch(searchFish(e.currentTarget.value))
})

export default connect(null, matchDispatchToProps)(Search)
