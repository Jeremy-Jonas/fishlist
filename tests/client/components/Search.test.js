import React from 'react'
import { mount, shallow } from 'enzyme'
import { Input } from 'semantic-ui-react'
import { searchFish } from '../../../client/actions/fish'
import { Search } from '../../../client/components/Search'


test('Initial test', () => {
  expect(1).toBe(1)
})

test('<Input /> in Search', () => {
  const expected = ''
  const wrapper = mount(<Input />)
  expect(wrapper.text()).toMatch(expected)
})


test('Search should return fish', () => {
  const expected = {"searchedFish": "smelly fish", "type": "SEARCH_FISH"}
  const actual = searchFish("smelly fish")
  expect(actual).toEqual(expected)
})

describe('Search', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Search debug />)

    expect(component).toMatchSnapshot()
  })
})

test('Search should render correctly with no props', () => {
  const component = shallow(<Search />)

  expect(component).toMatchSnapshot()
})