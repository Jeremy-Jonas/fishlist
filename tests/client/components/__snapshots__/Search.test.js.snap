// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Input
        icon="search"
        placeholder="Start typing to filter..."
        type="text"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "icon": "search",
        "onKeyUp": undefined,
        "placeholder": "Start typing to filter...",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Input
          icon="search"
          placeholder="Start typing to filter..."
          type="text"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "icon": "search",
          "onKeyUp": undefined,
          "placeholder": "Start typing to filter...",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Search should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Input
        icon="search"
        placeholder="Start typing to filter..."
        type="text"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "icon": "search",
        "onKeyUp": undefined,
        "placeholder": "Start typing to filter...",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Input
          icon="search"
          placeholder="Start typing to filter..."
          type="text"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "icon": "search",
          "onKeyUp": undefined,
          "placeholder": "Start typing to filter...",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
